### ISSUE TITLE
Unexpected JSON output of nested arrays of tables?

### ISSUE THREAD

**Issue by BjoernSchilberg** [Jonarod/markty-TOML#2]
Hej  @Jonarod,
I tried the nested array of tables example from
<https://github.com/toml-lang/toml#array-of-tables> with markty-TOML.

```javascript
var toml = require('markty-toml')

const someTOML = `
[[fruit]]
  name = "apple"

    [fruit.physical]
        color = "red"
            shape = "round"

  [[fruit.variety]]
    name = "red delicious"

  [[fruit.variety]]
    name = "granny smith"

[[fruit]]
  name = "banana"

  [[fruit.variety]]
    name = "plantain"
`

console.log( toml(someTOML) )
```

Unexpected JSON output:

```json
{ fruit: 
   [ { name: 'apple', physical: [Object], variety: [Array] },
     { name: 'banana', variety: [Array] },
     fruit: { name: 'banana', variety: [Array] } ] }
```

Would expect the following JSON output:
```json
{ fruit: 
   [ { name: 'apple', physical: [Object], variety: [Array] },
     { name: 'banana', variety: [Array] }
     ] }
```

Is my expectation wrong?  Did I miss something?

**Comment by Jonarod** [comment]
Sorry I may have missed something.

Do you get the same output using the live converter ?

Have a look here with your example: [https://jsfiddle.net/sL9ssuch/10/](https://jsfiddle.net/sL9ssuch/10/)

I do get the correct JSON output.

Could you confirm you get different results from the live converter and your code ?

P.S: the live converter actually implements the sample code you posted.

**Comment by BjoernSchilberg** [comment]
> Do you get the same output using the live converter ?
> Have a look here with your example: https://jsfiddle.net/sL9ssuch/10/
> I do get the correct JSON output.

Yes, with the live converter, the JSON output is displayed correct. 

The "magic" behind this seems to be the `JSON.stringify()` call. But if you left out the `JSON.stringify()`call, for example do an `marktytoml(TOML))` instead of  `JSON.stringify(marktytoml(TOML), null, 4)`  the JSON output differs. 

Look at the following example:  https://jsfiddle.net/mzh3qwvr/7/ and take a look in the output of your Browsers DevTools Console. Then you can see that the output from `console.log` differs from the `output.innerHTML` output.

**Comment by Jonarod** [comment]
Mmmm I see... Sorry for that.
I am traveling right now but I'll check it as soon as I come back.

**Comment by Jonarod** [comment]
Sorry for late reply, I have been quite busy these days.
Just released a patch with `v0.0.9`, give it a try here: [https://jsfiddle.net/sL9ssuch/13/](https://jsfiddle.net/sL9ssuch/13/)
